ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -419,115 +398,36 @@ public boolean hasSucceeded() {
-        && totalOfflineReplicaCount > 0) {
-      logger.info(
-          "Terminating {} on {} due to disk down count and total Not_Found count from eligible replicas and some "
-              + "other replicas being unavailable. CrossColoEnabled: {}, DiskDownCount: {}, TotalNotFoundCount: {}, "
-              + "TotalReplicaCount: {}, replicaSuccessTarget: {}, OfflineReplicaCount: {}, allReplicaCount: {} {} "
-              + "replicasByState = {}",
-          routerOperation, partitionId, crossColoEnabled, diskDownCount, totalNotFoundCount, totalReplicaCount,
-          replicaSuccessTarget, totalOfflineReplicaCount, allReplicaCount, getBlobIdLog(), allDcReplicasByState);
-      routerMetrics.failedMaybeDueToTotalOfflineReplicasCount.inc();
+      routerMetrics.failedOnTotalNotFoundCount.inc();
       return true;
     }
+
     return false;
   }
 
   @Override
-  public boolean hasFailedOnNotFound() {
-    if (routerOperation == RouterOperation.PutOperation) {
-      return false;
-    }
-    if (hasFailedOnOriginatingDcNotFound()) {
-      logger.info(
-          "Terminating {} on {} due to Not_Found failure. Originating Not_Found count: {}, failure threshold: {},"
-              + "originatingDcOfflineReplicaCount: {}, originatingDcNameTotalReplicaCount: {},"
-              + "replicaSuccessTarget: {}, allReplicaCount: {} {}", routerOperation.name(), partitionId,
-          originatingDcNotFoundCount, originatingDcNotFoundFailureThreshold, originatingDcOfflineReplicaCount,
-          originatingDcTotalReplicaCount, replicaSuccessTarget, allReplicaCount, getBlobIdLog());
-      routerMetrics.failedOnOriginatingDcNotFoundCount.inc();
-      return true;
-    }
-    // To account for GET operation, the threshold should be  >= totalReplicaCount - (success target - 1)
-    // Right now, this only applies for replica only partitions and may not be completely accurate if there are
-    // failures responses other than not found.
-    if (hasFailedOnCrossColoNotFound()) {
-      logger.info(
-          "Terminating {} on {} due to disk down count and total Not_Found. CrossColoEnabled: {}, DiskDownCount: {},"
-              + "TotalNotFoundCount: {}, TotalReplicaCount: {}, replicaSuccessTarget: {}, OfflineReplicaCount: {},"
-              + "allReplicaCount: {} {} replicasByState = {}", routerOperation, partitionId, crossColoEnabled, diskDownCount,
-          totalNotFoundCount, totalReplicaCount, replicaSuccessTarget, totalOfflineReplicaCount, allReplicaCount,
-          getBlobIdLog(), allDcReplicasByState);
-      routerMetrics.failedOnTotalNotFoundCount.inc();
-      return true;
-    }
-    return false;
+  public boolean hasFailedOnUnavailability() {
+    // If sufficient number of originating DC replicas are not available in leader or standby state, it means Ambry is
+    // unavailable.
+    return originatingDcName != null
+        && originatingDcLeaderOrStandbyReplicas.size() < originatingDcNotFoundFailureThreshold;
```

Comment:
please also check if there is any success count here.

Reply:
Sure. As we discussed offline, instead of checking success count, I updated `hasSomeUnavailability()` to return true if there is any replica in originating DC which is not in leader and standby state. Also added comments in java doc that this method can be used when operation has failed.